runtime/lua/vim/diagnostic.lua

```diff
@@ -890,22 +872,34 @@ M.handlers.signs = {
     end
 
     local ns = M.get_namespace(namespace)
-    if not ns.user_data.sign_group then
-      ns.user_data.sign_group = string.format('vim.diagnostic.%s', ns.name)
+    if not ns.user_data.sign_ns then
+      ns.user_data.sign_ns = api.nvim_create_namespace('')
     end
 
-    local sign_group = ns.user_data.sign_group
+    opts.signs.text = vim.tbl_extend(
+      'force',
+      vim
+        .iter({ unpack(M.severity, 1, 4) })
+        :map(function(v)
+          return v:sub(1, 1):upper()
+        end)
+        :totable(),
+      opts.signs.text or {}
+    )
+
     for _, diagnostic in ipairs(diagnostics) do
-      vim.fn.sign_place(0, sign_group, sign_highlight_map[diagnostic.severity], bufnr, {
+      local text = opts.signs.text[diagnostic.severity] or 'U'
+      pcall(api.nvim_buf_set_extmark, bufnr, ns.user_data.sign_ns, diagnostic.lnum, 0, {
```

Comment:
Why is `pcall` needed here?

Reply:
And if it is needed, its failure should be handled (by logging something, at least). pcall should (almost) never silently ignore failures (unless surrounding code handles that in some other way).

Reply:
because diagnostics include some buffers which not loaded. add a filter to remove pcall